In [1]:
#Import Modules
import pandas as pd

#Load CSV file
file = "purchase_data.csv"

#Read CSV File and store into Pandas data frame
pymoli_df = pd.read_csv(file)

In [2]:
#Player Count
#Find number of unique screen names
total_players = pymoli_df["SN"].nunique()

#display total number of players in dataframe
total_players_df = pd.DataFrame({"Total Players" :[total_players]})

total_players_df


Total Players
0            576

In [3]:
#Purchasing analysis (Total): finding values
total_uitems = pymoli_df["Item ID"].nunique()
avg_price = pymoli_df['Price'].mean()
num_purchase = pymoli_df["Purchase ID"].count()
total_rev = pymoli_df["Price"].sum()


In [4]:
#Purchasing Analysis (Total): Creating dictionary 
purchase_dict = {
                    "Number of Unique Items" : [(total_uitems)], "Average Price" : [(avg_price)],
                    "Number of Purchases" : [(num_purchase)], "Total Revenue" : [(total_rev)]}

#Create Dataframe with formatting
purchase_df = pd.DataFrame(purchase_dict)

purchase_df["Average Price"]= purchase_df["Average Price"].map("${:.2f}".format)
purchase_df["Total Revenue"] = purchase_df["Total Revenue"].map("${:,.2f}".format)

purchase_df


Number of Unique Items Average Price  Number of Purchases Total Revenue
0                     179         $3.05                  780     $2,379.77

In [5]:
#Gender Demographics
#Grouping by Gender
gender_group_df = pymoli_df.groupby(["Gender"])

#Creating dataframe
gender_df = pd.DataFrame()

gender_df['Total Count'] = gender_group_df['SN'].nunique()
gender_df['Percentage of Players'] = gender_df['Total Count']/total_players *100

#format column 
gender_df["Percentage of Players"]= gender_df["Percentage of Players"].map("{:.2f}%".format)

#Sort high to low
gen_order = gender_df.sort_values( 'Total Count', ascending = False)

gen_order







Total Count Percentage of Players
Gender                                                  
Male                           484                84.03%
Female                          81                14.06%
Other / Non-Disclosed           11                 1.91%

In [6]:
#Purchasing Analysis (Gender)
#Still using groupby gender from above line

#Creating dataframe
gen_pur_df = pd.DataFrame()

gen_pur_df["Purchase Count"] = gender_group_df["Purchase ID"].count()
gen_pur_df["Average Purchase Price"] = gender_group_df["Price"].mean()
gen_pur_df["Total Purchase Price"] = gender_group_df["Price"].sum()
gen_pur_df["Avg Total Purchase per Person"] = gender_group_df["Price"].sum() / gender_df['Total Count']

#Format columns
gen_pur_df["Average Purchase Price"]= gen_pur_df["Average Purchase Price"].map("${:.2f}".format)
gen_pur_df["Total Purchase Price"]= gen_pur_df["Total Purchase Price"].map("${:,.2f}".format)
gen_pur_df["Avg Total Purchase per Person"]= gen_pur_df["Avg Total Purchase per Person"].map("${:.2f}".format)

gen_pur_df


Purchase Count Average Purchase Price  \
Gender                                                         
Female                            113                  $3.20   
Male                              652                  $3.02   
Other / Non-Disclosed              15                  $3.35   

                      Total Purchase Price Avg Total Purchase per Person  
Gender                                                                    
Female                             $361.94                         $4.47  
Male                             $1,967.64                         $4.07  
Other / Non-Disclosed               $50.19                         $4.56

In [7]:
#Age Demographics
#Binning

# Create bins 
bins = [0,10,15,20,25,30,35,40,150]

# Create labels for these bins
group_labels = ["<10", "10-14", "15-19", "20-24", "25-29", "30-34",
                "35-39", "40+"]


In [8]:
#Age Demographics
#Add column with bins to dataframe

pymoli_df ['Age Range'] = pd.cut(pymoli_df['Age'], bins, labels = group_labels, include_lowest = True, right = False)

#Group by newly created bins of Age Range
age_group_df = pymoli_df.groupby(["Age Range"])

#Create dataframe
age_df = pd.DataFrame()
age_df['Total Count'] = age_group_df['SN'].nunique()
age_df['Percentage of Players'] = age_df['Total Count']/total_players *100

#format column
age_df["Percentage of Players"]= age_df["Percentage of Players"].map("{:.2f}%".format)

age_df

Total Count Percentage of Players
Age Range                                   
<10                 17                 2.95%
10-14               22                 3.82%
15-19              107                18.58%
20-24              258                44.79%
25-29               77                13.37%
30-34               52                 9.03%
35-39               31                 5.38%
40+                 12                 2.08%

In [9]:
#Purchasing Analysis (Age)
#Still using groupby Age Range from above line

#Create Dataframe
age_pur_df = pd.DataFrame()

age_pur_df["Purchase Count"] = age_group_df["Purchase ID"].count()
age_pur_df["Average Purchase Price"] = age_group_df["Price"].mean()
age_pur_df["Total Purchase Price"] = age_group_df["Price"].sum()
age_pur_df["Avg Total Purchase per Person"] = age_group_df["Price"].sum() / age_df['Total Count']

#Format columns
age_pur_df["Average Purchase Price"]= age_pur_df["Average Purchase Price"].map("${:.2f}".format)
age_pur_df["Total Purchase Price"]= age_pur_df["Total Purchase Price"].map("${:,.2f}".format)
age_pur_df["Avg Total Purchase per Person"]= age_pur_df["Avg Total Purchase per Person"].map("${:.2f}".format)

age_pur_df

Purchase Count Average Purchase Price Total Purchase Price  \
Age Range                                                               
<10                    23                  $3.35               $77.13   
10-14                  28                  $2.96               $82.78   
15-19                 136                  $3.04              $412.89   
20-24                 365                  $3.05            $1,114.06   
25-29                 101                  $2.90              $293.00   
30-34                  73                  $2.93              $214.00   
35-39                  41                  $3.60              $147.67   
40+                    13                  $2.94               $38.24   

          Avg Total Purchase per Person  
Age Range                                
<10                               $4.54  
10-14                             $3.76  
15-19                             $3.86  
20-24                             $4.32  
25-29                             $3.81  
30-34                             $4.12  
35-39                             $4.76  
40+                               $3.19

In [10]:
#Top Spenders

#Group by Screen Name
sn_group_df = pymoli_df.groupby(["SN"])

#Create dataframe
spend_df = pd.DataFrame()

spend_df["Purchase Count"] = sn_group_df["SN"].count()
spend_df["Average Purchase Price"] = sn_group_df["Price"].mean()
spend_df["Total Purchase Value"] = sn_group_df["Price"].sum()

#Find 5 top spenders
high_spend_df = spend_df.nlargest(5,"Total Purchase Value")


#Format columns
high_spend_df["Average Purchase Price"]= high_spend_df["Average Purchase Price"].map("${:.2f}".format)
high_spend_df["Total Purchase Value"]= high_spend_df["Total Purchase Value"].map("${:,.2f}".format)


high_spend_df

Purchase Count Average Purchase Price Total Purchase Value
SN                                                                     
Lisosia93                 5                  $3.79               $18.96
Idastidru52               4                  $3.86               $15.45
Chamjask73                3                  $4.61               $13.83
Iral74                    4                  $3.40               $13.62
Iskadarya95               3                  $4.37               $13.10

In [11]:
#Most Popular Items
#Group by Item ID and Item Name
item_group_df = pymoli_df.groupby(["Item ID","Item Name"])

#Create dataframe
item_df = pd.DataFrame()

item_df["Purchase Count"] = item_group_df["SN"].count()
item_df["Item Price"] = item_group_df["Price"].sum() / item_df['Purchase Count']
item_df["Total Purchase Value"] = item_group_df["Price"].sum()

#Find the 5 most popular items
high_item_df = item_df.nlargest(5,"Purchase Count", keep = "last")


#Format columns
high_item_df["Item Price"]= high_item_df["Item Price"].map("${:.2f}".format)
high_item_df["Total Purchase Value"]= high_item_df["Total Purchase Value"].map("${:,.2f}".format)


high_item_df

Purchase Count  \
Item ID Item Name                                                      
92      Final Critic                                              13   
178     Oathbreaker, Last Hope of the Breaking Storm              12   
145     Fiery Glass Crusader                                       9   
132     Persuasion                                                 9   
108     Extraction, Quickblade Of Trembling Hands                  9   

                                                     Item Price  \
Item ID Item Name                                                 
92      Final Critic                                      $4.61   
178     Oathbreaker, Last Hope of the Breaking Storm      $4.23   
145     Fiery Glass Crusader                              $4.58   
132     Persuasion                                        $3.22   
108     Extraction, Quickblade Of Trembling Hands         $3.53   

                                                     Total Purchase Value  
Item ID Item Name                                                          
92      Final Critic                                               $59.99  
178     Oathbreaker, Last Hope of the Breaking Storm               $50.76  
145     Fiery Glass Crusader                                       $41.22  
132     Persuasion                                                 $28.99  
108     Extraction, Quickblade Of Trembling Hands                  $31.77

In [12]:
#Most Profitable Items
#Still using Group by Item ID and Item Name from above line

#Create dataframe
item_df = pd.DataFrame()

item_df["Purchase Count"] = item_group_df["SN"].count()
item_df["Item Price"] = item_group_df["Price"].sum() / item_df['Purchase Count']
item_df["Total Purchase Value"] = item_group_df["Price"].sum()

#Find the 5 most profitable items
high_item_df = item_df.nlargest(5,"Total Purchase Value", keep = "last")


#Format columns
high_item_df["Item Price"]= high_item_df["Item Price"].map("${:.2f}".format)
high_item_df["Total Purchase Value"]= high_item_df["Total Purchase Value"].map("${:,.2f}".format)


high_item_df

Purchase Count  \
Item ID Item Name                                                      
92      Final Critic                                              13   
178     Oathbreaker, Last Hope of the Breaking Storm              12   
82      Nirvana                                                    9   
145     Fiery Glass Crusader                                       9   
103     Singed Scalpel                                             8   

                                                     Item Price  \
Item ID Item Name                                                 
92      Final Critic                                      $4.61   
178     Oathbreaker, Last Hope of the Breaking Storm      $4.23   
82      Nirvana                                           $4.90   
145     Fiery Glass Crusader                              $4.58   
103     Singed Scalpel                                    $4.35   

                                                     Total Purchase Value  
Item ID Item Name                                                          
92      Final Critic                                               $59.99  
178     Oathbreaker, Last Hope of the Breaking Storm               $50.76  
82      Nirvana                                                    $44.10  
145     Fiery Glass Crusader                                       $41.22  
103     Singed Scalpel                                             $34.80

Heroes of Pymoli: Conclusions from Data Analysis

Erik Hoaglund

Written Report HW 4

NU Bootcamp

7/22/21

1. Conclusions

    A. Breaking down the purchasing behavior by gender, while males are much more likely to make in game purchases compared to females, females per person spend on average 40 cents more than males per person.
    
    B. Breaking down the players by age range, the 20-24 age range is by far the most popular age range making in game purchases with 258 people making purchases, more than doubling the next closest age range of 15-19 (107 people making purchases).
    
    C. While the 20-24 age range makes the most purchases, they do not spend the most per person. The 35-39 age range spends the most per person with four dollars and 76 cents. They are followed closely behind by the under 10 age range who spends four dollars and 54 cents per person.